spynnaker/pyNN/models/neuron/population_machine_neurons.py

```diff
@@ -0,0 +1,230 @@
+# Copyright (c) 2017-2020The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+import ctypes
+from collections import namedtuple
+
+from spinn_utilities.abstract_base import abstractproperty, abstractmethod
+from spinn_utilities.overrides import overrides
+
+from spinn_front_end_common.utilities.utility_objs import ProvenanceDataItem
+from spinn_front_end_common.interface.provenance import (
+    ProvidesProvenanceDataFromMachineImpl)
+from spynnaker.pyNN.models.abstract_models import (
+    AbstractReadParametersBeforeSet)
+from spynnaker.pyNN.utilities.constants import SPIKE_PARTITION_ID
+from spinn_front_end_common.utilities import helpful_functions
+
+
+get_placement_details = \
+    ProvidesProvenanceDataFromMachineImpl._get_placement_details
+add_name = ProvidesProvenanceDataFromMachineImpl._add_name
+
+
+class NeuronProvenance(ctypes.LittleEndianStructure):
+    """ Provenance items from neuron processing
+    """
+    _fields_ = [
+        # The timer tick at the end of simulation
+        ("current_timer_tick", ctypes.c_uint32),
+        # The number of misses of TDMA time slots
+        ("n_tdma_misses", ctypes.c_uint32)
+    ]
+
+    N_ITEMS = len(_fields_)
+
+
+# Identifiers for neuron regions
+NeuronRegions = namedtuple(
+    "NeuronRegions",
+    ["neuron_params", "neuron_recording"])
+
+
+class PopulationMachineNeurons(AbstractReadParametersBeforeSet):
```

Comment:
id think this is a abstract as well. it makes abstract properities afterall. thus cant be built itself

Reply:
As above - inherits from abstract so abstraction will be checked